### PR TITLE
ATF-9 Create AtomFeed server database & implement logic that write events to the database

### PR DESCRIPTION
**Pull request by dserkowski**
What have been done:
* Create database for AtomFeed structures
* Add logic in order to save events' data (implement EventManager.class logic)
* Added FeedWriter interface & DefaultFeedWriter
* Added logs
* Some corrections in the module

More information [under the ticket](https://issues.openmrs.org/browse/ATF-9)

### CODE REVIEW

File: api/src/main/resources/defaultAtomfeedConfig.json [hunk 1]
```diff
@@ -1,13 +1,169 @@
 [
   {
+    "openMrsClass": "org.openmrs.Location",
+    "enabled": "false",
+    "title": "Location",
+    "category": "location",
+    "linkTemplates": {
+      "rest": "openmrs/ws/rest/v1/location/{uuid}?v=full",
+      "fhir": "openmrs/ws/fhir/v1/location/{uuid}?v=full"
```
Comment (pkornowski) [inline api/src/main/resources/defaultAtomfeedConfig.json]: Link template for REST API is configured properly. However, FHIR API has gently another template. In every resource template should looks like openmrs/ws/fhir/Location/{uuid}
- resource should start with capital Letter ( i.e. Location here)
- v1 and ?v=full should disappear from the template

Only in that case templates give us possibility to get resources from FHIR

File: api/src/main/java/org/openmrs/module/atomfeed/api/writers/impl/DefaultFeedWriter.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+				getEventContent(openmrsObject, feedConfiguration),
+				feedConfiguration.getCategory(),
+				eventAction.name()
+		);
+		debugEvent(event);
+		saveEvent(event);
+		LOGGER.info("A feed for {} has been saved in AtomFeed", openmrsObject.getClass().getName());
+	}
+	
+	private String getEventContent(OpenmrsObject openmrsObject, FeedConfiguration feedConfiguration) {
```
Comment (pkornowski) [inline api/src/main/java/org/openmrs/module/atomfeed/api/writers/impl/DefaultFeedWriter.java]: Currently, you choose preferred link template and then its saved to the database. I think there should be saved every template from the defined configuration. The Sync module should decide which of these would like to read and process.

Comment (dserkowski) [reply api/src/main/java/org/openmrs/module/atomfeed/api/writers/impl/DefaultFeedWriter.java]: Ok, I will fix it

File: api/src/main/resources/defaultAtomfeedConfig.json (hunk 1 above)
Comment (dserkowski) [reply api/src/main/resources/defaultAtomfeedConfig.json]: Ok, I will fix it

**Comment by pkornowski** [comment]
Now everything works correctly. Thanks.

**Pull request merged**